woocommerce/wc-admin
# Highlight help panel with tooltip

Fixes #5082 

Added the `woocommerce_task_list_tracked_started_tasks` option as an object of { task_name: task_visit_count } to keep track of the tasks that where visited, and the `woocommerce_help_panel_highlight_shown` option to manage if the help panel tooltip was shown. The task_visit_count is increased everytime the task is clicked from the task list.

The tooltip will show when it is the second time an uncompleted task has been visited. The tooltip will **only** be hidden if a user has interacted with it. If a user refreshes without interacting it will show the tooltip again.

The tooltip is shown using a HighlightTooltip component, it makes use of an element id to target, and makes use of the wordpress `createPortal` function  and `Popover` component. This way it acts similar to the [Woo admin pointers](https://github.com/woocommerce/woocommerce/blob/master/includes/admin/class-wc-admin-pointers.php#L228).

### Accessibility


### Screenshots

![helppanel-popup](https://user-images.githubusercontent.com/2240960/101396426-b0152d00-38a1-11eb-8a6f-a0feac22fa2b.gif)

### Detailed test instructions:

- Create a new WooCommerce store, and finish the initial Wizard
- Navigate to one of the unfinished tasks on the home screen and finish it (Ex: `Add my products` and add a product)
- Go to home screen and click on the finished task (no popup should appear)
- Go to home screen and click an unfinished task (not finishing it)
- Go to home screen again and click the same unfinished task
- A popup should appear now, highlighting the `Help` panel, like this:
<img width="352" alt="Screen Shot 2020-12-07 at 3 39 42 PM" src="https://user-images.githubusercontent.com/2240960/101396996-798be200-38a2-11eb-8345-53a904460bbb.png">
- Click `Got it` or the close button.
- Help panel button should still work correctly.
- If navigating back home and to the unfinished task again the popup should not show up (this goes for any other unfinished tasks as well).
- It should of logged the `wcadmin_help_tooltip_view` when the tooltip was viewed, and the `wcadmin_help_tooltip_click` when it was dismissed.
